### PR TITLE
Fixes autoloading, plugins and tests on Windows

### PR DESCRIPTION
**Pull request by dimitrioskarvounaris**
Based on https://github.com/pestphp/pest/pull/41 and all tests for `test:unit` pass under Windows and Linux.

It fixes issues, where otherwise some features wouldn't load or work at all under Windows.

### CODE REVIEW

**Comment by nunomaduro** [comment]
Sounds a fantastic job. Do you know why tests are failing?

File: tests/Visual/SingleTestOrDirectory.php [hunk 1]
```diff
@@ -3,63 +3,54 @@
...
 };
 
 test('allows to run a single test', function () use ($run) {
-    assertStringContainsString(<<<EOF
-   PASS  Tests\Fixtures\DirectoryWithTests\ExampleTest
-  ✓ it example
-
-  Tests:  1 passed
-EOF, $run('tests/Fixtures/DirectoryWithTests/ExampleTest.php'));
+    $output = $run('tests/Fixtures/DirectoryWithTests/ExampleTest.php');
```
Comment (nunomaduro) [inline tests/Visual/SingleTestOrDirectory.php:14]: Can you revert this changes? I prefer in the way it was before.

If don't work on Windows, but create a file with the contents and do a `file_get_contents` on it.

Comment (dimitrioskarvounaris) [reply tests/Visual/SingleTestOrDirectory.php:14]: Not sure how you mean with the file_get_contents()?
I were going to solve it differently first, by replacing the \r\n with \n, but then I understood, that there are cases, where anything that still contains linebreaks can break in more than one way and as such the line breaks should not be part of the comparison. Having a file, could still make it fail in some cases I think?

**Comment by dimitrioskarvounaris** [comment]
> Do you know why tests are failing?

You mean, if I know, why the tests were failing? Yes, I debugged the code and figured the reason for each one. Some things are posted in Discord. 

Often - besides the linebreaks - it was that paths would not match, and as such, working on strings where part of path would be replaced or something added, would no longer happen or not happen correctly. Especially, when $rootPath was used. This would even lead to things as having the wrong output next to "PASS", traits not autoloading or plugins not even initializing with the "invalid path exception" when Plugin::uses() (as it sets `->in($rootPath)`) was used.

File: tests/Visual/SingleTestOrDirectory.php (hunk 1 above)
Comment (dimitrioskarvounaris) [reply tests/Visual/SingleTestOrDirectory.php:14]: I know, I like the other way it was also better, but this one works always.

Comment (nunomaduro) [reply tests/Visual/SingleTestOrDirectory.php:19]: Can you revert all the changes in SingleTestOrDirectory.php ? Except the line 6.

File: src/PendingObjects/UsesCall.php [hunk 2]
```diff
@@ -68,12 +78,12 @@ public function in(string ...$targets): void
         }, $targets);
 
         $this->targets = array_map(function ($target): string {
-            $realTarget = realpath($target);
-            if ($realTarget === false) {
-                throw new InvalidUsesPath($target);
+            $isValid = is_dir($target) || file_exists($target);
+            if (!$isValid) {
+                throw new InvalidUsesPath($target . "\n");
```
[the remark this answers is not in the document]
Comment (dimitrioskarvounaris) [reply src/PendingObjects/UsesCall.php:83]: No idea. I think it wasn't from me... can be removed I guess

**Comment by nunomaduro** [comment]
@dimitrioskarvounaris @Pierstoval Huge thanks! <3
